align.go

```diff
@@ -0,0 +1,115 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plot
+
+import (
+	"math"
+
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+)
+
+// Align returns a two-dimensional row-major array of Canvases which will
+// produce tiled plots with DataCanvases that are evenly sized and spaced.
+// The arguments to the function are a two-dimensional row-major array
+// of plots, a tile configuration, and the canvas to which the tiled
+// plots are to be drawn.
+func Align(plots [][]*Plot, t draw.Tiles, dc draw.Canvas) [][]draw.Canvas {
```

Comment:
If I understand right, the main reason for doing this would be to avoid doing the final nested loop to draw the plots to the canvases, and also probably an additional nested loop to create the original matrix of plots. I'm guessing that @kortschak would make a "weight of things" argument that the added convenience would not be worth the added API surface. 

Another option could be to take a 1-d slice of plots and return a 1-d array of canvases, which would mean that the user wouldn't have to go through the nested loops, but they would have to do the 1-d to 2-d array index conversion on their own. I'm not sure whether that would make things easier overall or not. My general feeling is that array index conversions tend to be more confusing than nested loops.

I do like the idea of the idea of the TiledPlot, though, or maybe something like a "MultiPlotter" interface that could allow the automatic creation of things like the covariance matrices you linked to. My feeling is that that would be something that would be built on top of this PR rather than built into it, but let me know what you think.